Slim/Player/Squeezebox.pm

```diff
@@ -70,31 +70,35 @@ sub reconnect {
 
 	my $controller = $client->controller();
 
-	if (!$reconnect) {
+	if (!defined $reconnect) {
+		
+		# reconnection of a forgotten client, need to take resume position from prefs
+		$client->resumeOnPower(1) if $client->power();
+		
+	} else {
 
 		if ($client->power()) {
+			$client->resumeOnPower();
 			$controller->playerActive($client);
 		}
 
-		if ($controller->onlyActivePlayer($client)) {
-			main::INFOLOG && $sourcelog->is_info && $sourcelog->info($client->id . " restaring play on pseudo-reconnect at "
-				. ($bytes_received ? $bytes_received : 0));
-			$controller->playerReconnect($bytes_received);
-		}
+		# disconnected but not forgotten clients may need a restart or a proper stop
+		if (!$reconnect) {
+			if ($controller->onlyActivePlayer($client)) {
+				main::INFOLOG && $sourcelog->is_info && $sourcelog->info($client->id . " restarting play on pseudo-reconnect at "
+					. ($bytes_received ? $bytes_received : 0));
+				$controller->playerReconnect($bytes_received);
+			}
 
-		if ($client->isStopped()) {
-			# Ensure that a new client is stopped, but only on sb2s
-			if ( $client->isa('Slim::Player::Squeezebox2') ) {
-				main::INFOLOG && $sourcelog->is_info && $sourcelog->info($client->id . " forcing stop on pseudo-reconnect");
-				$client->stop();
+			if ($client->isStopped()) {
+				# Ensure that a new client is stopped, but only on sb2s
+				if ( $client->isa('Slim::Player::Squeezebox2') ) {
+					main::INFOLOG && $sourcelog->is_info && $sourcelog->info($client->id . " forcing stop on pseudo-reconnect");
+					$client->stop();
+				}
 			}
-		}
-	} else {
-		# bug 16881: player in a sync-group may have been made inactive upon disconnect;
-		# make sure it is active now.
-		if ($client->power()) {
-			$controller->playerActive($client);
```

Comment:
Did you test the sync group case? I don't see `$controller->playerActive()` being called in the new implementation.

Reply:
I will check but I remember it being called by the sync code itself. I don't think it was called before, was it?

Reply:
It should have been called but I thought the Sync::restoreSync above would do the job. In fact, if you see my other command in Client::startup, I don't sync restoreSync should be there and that explains why the Squeezebox::reconnect call to restoreSync does not do the job (the sync has been pseudo-restored but players might not be connected yet) and that playerActive must be "manually" called.